skbio/tree/_tree.py

```diff
@@ -3098,3 +3098,33 @@ def shuffle(self, k=None, names=None, shuffle_f=np.random.shuffle, n=1):
 
             yield self
             counter += 1
+
+    def _cache_leafcount(self):
```

Comment:
It is not clear to me whether these should be executed on the `create_caches`, but it perhaps makes sense.